app/models/posts.js

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export class Posts {
+    get propsAsJson() {
+        try {
+            return JSON.parse(this.props);
+        } catch {
+            return null;
+        }
+    }
+
+    set propsFromJson(props) {
+        this.props = JSON.stringify(props);
+    }
+
+    static schema = {
+        name: 'Posts',
+        primaryKey: 'id',
+        properties: {
+            id: 'string',
+            createAt: {type: 'int', indexed: true},
+            updateAt: {type: 'int', indexed: true},
+            deleteAt: {type: 'int', default: 0},
+            editAt: {type: 'int', default: 0},
+            user: 'Users',
+            rootId: 'string?',
+            originalId: 'string',
+            pendingPostId: 'string?',
+            message: {type: 'string', default: ''},
+            type: 'string?',
+            props: 'string',
+            files: 'Files[]',
+            hasReactions: {type: 'bool', default: false},
+            isPinned: {type: 'bool', default: false},
+            metadata: 'Metadata[]',
+            reactions: 'Reactions[]',
+        },
+    }
+}
+
+export class Metadata {
```

Comment:
I think you might have the wrong object here since this structure is the metadata for a link which is never sent to the client.

If this is intended, it might be better to refer to this as `LinkMetadata` to avoid potential naming clashes in the future

Reply:
I'll take another look to the metadata structure, cause that's what we want here